990 introduce support for the tdls library (#992)
thelfer/tfel: docs/web/release-notes-5.2.md

The page describes the new functionalities of Version 5.2 of the
`TFEL` project.

# Highlights

> ## Support for the `TDLS` (`TinyDeviceCalledLinearSolver`)

Reviewer: ```suggestion
## Support for the `TDLS` (`TinyDeviceCallableLinearSolver`)
```

> The `TDLS` library provides an highly optimized linear solver on `CPU`s

Reviewer: ```suggestion
The `TDLS` library provides tiny linear solvers for small dense systems,
callable on `CPU`s as well as inside `GPU` kernels, and designed for
maximum GPU performance:
```

> and `GPU`s:

Reviewer: ```suggestion
```

shows how to use it in the implementation of Ponte-Castaneda variational
bound (1992), within the framework of non-linear elasticity.

> <<<<<<< HEAD

Reviewer: ```suggestion
```

> # New features in `MFront`

Reviewer: ```suggestion
```

  `first_acceptable` (or equivalently `first-acceptable` or
  `FirstAcceptable`) or `full_scan` (or equivalently `full-scan` or
  `FullScan`).
- unroll_inner_loop: expected values are either `true` or `false`. If
  `true`, loops indexing register tiles carry a forced-unroll pragma -
  the guard that keeps tiles in registers on GPU backends, where a
  rolled loop indexes the tile dynamically and demotes it to slow local
  memory. If `false`, no unroll pragma anywhere - faster compiles, GPU
  performance not guaranteed.

If a parameter is not defined, the default values are inherited for the
`TiledLUppDefaultConfig` class defined by the `TDLS` library. See the
[API reference](https://trsxvz.github.io/TDLS/api/index.html) for
details. At the time of writing this documentation, those default values
are:

- `3` for `tile_size`
- `right_looking` for `schedule`
- `first_acceptable` for `out_of_tile_search_strategy`
- `true` for `unroll_inner_loop`

### Examples of usage

~~~~ {#LinearSystemSolver .cpp}
@LinearSystemSolver TDLS{
  tile_size: 4,
  unroll_inner_loop: false
};
~~~~

~~~~ {#LinearSystemSolver .cpp}
@LinearSystemSolver TDLS{
  tile_size: "StensorSize"
};
~~~~

> =======

Reviewer: ```suggestion
```

> # New `MFront` features

Reviewer: ```suggestion
```

#### `CUDA`

> >>>>>>> master

Reviewer: ```suggestion
```
